### PR TITLE
Bugfix of incorrect movement of inline comments

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Fix logic of moving long inline comments
* Expand tests
* Add utilities for processing the nodes on specific line

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt [hunk 1]
```diff
@@ -729,6 +729,41 @@ fun ASTNode.lastLineNumber() = getLineNumber() + text.count { it == '\n' }
...
+ * @param line
+ * @return corresponding list of nodes
+ */
+fun ASTNode.findAllNodesOnLine(
+    line: Int
+): List<ASTNode>? {
+    val rootNode = this.getRootNode()
+    val fileLines = rootNode.text.lines()
+
+    if (line <= 0 || line > fileLines.size - 1) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:744]:
```suggestion
    if (line !in fileLines.indices) {
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt [hunk 2]
```diff
@@ -729,6 +729,41 @@ fun ASTNode.lastLineNumber() = getLineNumber() + text.count { it == '\n' }
...
+}
+
+/**
+ * Return all nodes located at the specific line satisfying [condition]
+ *
+ * @param line
+ * @param condition
+ * @return corresponding list of nodes
+ */
+fun ASTNode.findAllNodesWithSpecificTypeOnLine(
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:762]:
```suggestion
fun ASTNode.findAllNodesWithConditionOnLine(
```

**Review by petertrr — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt (hunk 2 above)
Comment (kgevorkyan) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt]: Yes, it's more appropriate

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt (hunk 1 above)
Comment (kgevorkyan) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt]: But numeration of lines starts from 1, so we need return `null` for `line == 0` too, however I also noticed the another defect in this conditions, for the same reason as before, line could have the value `= fileLines.size`

**Pull request merged**